auto_nag/scripts/topcrash_add_keyword.py

```diff
@@ -0,0 +1,175 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import re
+
+from libmozdata.bugzilla import Bugzilla
+from libmozdata.connection import Connection
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.topcrash import Topcrash
+
+# TODO: should be moved when resolving https://github.com/mozilla/relman-auto-nag/issues/1384
+HIGH_SEVERITY = {"S1", "critical", "S2", "major"}
+
+MAX_SIGNATURES_PER_QUERY = 30
+SIGNATURES_PAT = re.compile(r"\[@ ?(.*?) ?\]")
+
+
+class TopcrashAddKeyword(BzCleaner):
+    def description(self):
+        return "Bugs with topcrash keyword but incorrect severity"
+
+    def columns(self):
+        return ["id", "summary", "severity", "added_keyword"]
+
+    @staticmethod
+    def __get_mail_to_ni(bug):
+        for field in ["assigned_to", "triage_owner"]:
+            person = bug.get(field, "")
+            if not utils.is_no_assignee(person):
+                return {"mail": person, "nickname": bug[f"{field}_detail"]["nick"]}
+
+        return None
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        if bugid in data:
+            return
+
+        has_topcrash_keyword = "topcrash" in bug["keywords"]
+        has_startup_keyword = "topcrash-startup" in bug["keywords"]
+        if has_topcrash_keyword and has_startup_keyword:
+            return
+
+        signatures = SIGNATURES_PAT.findall(bug["cf_crash_signature"])
```

Comment:
There is already:
https://github.com/mozilla/relman-auto-nag/blob/0103bd251962b25a27de0bcd69d1569bd5a366fc/auto_nag/utils.py#L111

which of course can be improved with regexp or whatever.

Reply:
Thank you! Done in 68dc395620068338f24ff047395d7de27865cb33.

I dropped the regexp implementation since it is less efficient. The following is a simple benchmark:
```
current: 89.97726325
regexp 229.680521334
```